### PR TITLE
CLOUDP-191823: Push to quay

### PR DESCRIPTION
**Pull request by nicohrubec**
## Proposed changes

Updates the docker release pipeline to also push to quay in addition to dockerhub. (Same frequency, same tagging strategy)

Here is a test run of the updated pipeline: https://github.com/mongodb/mongodb-atlas-cli/actions/runs/5714843666/job/15482960226

_Jira ticket:_ CLOUDP-191823

Closes CLOUDP-191823

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .github/workflows/docker-release.yml [hunk 1]
```diff
@@ -40,13 +41,22 @@ jobs:
...
         with:
           context: .
           platforms: ${{ env.PLATFORMS }}
           tags: ${{ env.IMAGE_REPOSITORY }}:latest ,
             ${{ env.IMAGE_REPOSITORY }}:${{ env.LATEST_VERSION }} ,
-            ${{ env.IMAGE_REPOSITORY }}:${{ env.LATEST_VERSION }}-${{ env.DATE }}
+            ${{ env.IMAGE_REPOSITORY }}:${{ env.LATEST_VERSION }}-${{ env.DATE }} ,
+            ${{ env.QUAY }}/${{ env.IMAGE_REPOSITORY }}:latest ,
+            ${{ env.QUAY }}/${{ env.IMAGE_REPOSITORY }}:${{ env.LATEST_VERSION }} ,
+            ${{ env.QUAY }}/${{ env.IMAGE_REPOSITORY }}:${{ env.LATEST_VERSION }}-${{ env.DATE }}
```
Comment (matt-condon) [inline .github/workflows/docker-release.yml:60]: could you make pushing to quay a separate step to pushing to dockerhub?

Comment (nicohrubec) [reply .github/workflows/docker-release.yml:60]: Yes, one possibility would be [this](https://docs.docker.com/build/ci/github-actions/copy-image-registries/). Not sure if this is cleaner though.

Alternatively I could add a second build and push step, which would require rebuilding the image a second time if we are okay with that.

Comment (matt-condon) [reply .github/workflows/docker-release.yml:60]: > Yes, one possibility would be [this](https://docs.docker.com/build/ci/github-actions/copy-image-registries/).
This approach looks good to me

Comment (nicohrubec) [reply .github/workflows/docker-release.yml]: Updated

**Review by matt-condon — APPROVED** [review]
LGTM, thanks for making that change!

**Pull request merged**